lib/elixir/lib/uri.ex

```diff
@@ -204,9 +208,9 @@ defmodule URI do
       true
 
   """
-  @spec char_reserved?(term) :: boolean
-  def char_reserved?(c) do
-    c in ':/?#[]@!$&\'()*+,;='
+  @spec char_reserved?(pos_integer) :: boolean
+  def char_reserved?(char) when is_integer(char) and char > 0 do
```

Comment:
well,i think you don't want `pos_integer` but `char` here.
And you can check it by doing ` def char_reserved?(char) when char in 0..0x10ffff do`